cmd/distrogen/main.go

```diff
@@ -104,12 +109,16 @@ func generateDistribution() error {
 		}
 		return err
 	}
-	if err := generator.MoveGeneratedDirToWd(); err != nil {
-		if err := generator.Clean(); err != nil {
-			fmt.Printf("couldn't clean generated dir: %v\n", err)
-		}
-		return err
+
+	var resultErr error
+	if *flagCompare {
+		resultErr = generator.Compare()
+	} else {
+		resultErr = generator.MoveGeneratedDirToWd()
 	}
 
-	return nil
+	if err := generator.Clean(); err != nil {
+		fmt.Printf("couldn't clean generated dir: %v\n", err)
+	}
```

Comment:
Your using `clean()` on line 106, maybe refactor into a `defer`?

Reply:
Done in [6ffda1a](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/279/commits/6ffda1a6013167880b780d05ca0279628dbb6615)